### PR TITLE
Default distro version for tests

### PR DESCRIPTION
**Pull request by danmosemsft**
Fix https://github.com/dotnet/corefx/issues/38443

I removed VERSION_ID from my etc/os-release and verified it fails without this change, passes with it.

### CODE REVIEW

**Comment by danmosemsft** [comment]
Hmm, maybe this should go into NormalizeDistroInfo.

**Comment by danmosemsft** [comment]
OK, should be good now. I managed to throughly confuse myself by pushing from cmd rather than WSL.

File: src/CoreFx.Private.TestUtilities/src/System/PlatformApis.Unix.cs [hunk 1]
```diff
@@ -203,12 +203,13 @@ private static DistroInfo LoadDistroInfo()
                     }
                 }
             }
+            // possibly should fall back to /usr/lib/os-release here
```
Comment (stephentoub) [inline src/CoreFx.Private.TestUtilities/src/System/PlatformApis.Unix.cs]: What is this comment referring to?  A future todo?  The code that follows?  What's the action item here?

File: src/CoreFx.Private.TestUtilities/src/System/PlatformApis.Unix.cs [hunk 2]
```diff
@@ -236,6 +237,13 @@ private static DistroInfo NormalizeDistroInfo(DistroInfo distroInfo)
                 distroInfo.VersionId = distroInfo.VersionId.Substring(0, lastVersionNumberSeparatorIndex);
             }
 
+            // In some distros/versions we cannot discover the distro version; return something valid.
+            // Pick a high version number, since this seems to happen on newer distros.
+            if (String.IsNullOrEmpty(distroInfo.VersionId))
```
Comment (stephentoub) [inline src/CoreFx.Private.TestUtilities/src/System/PlatformApis.Unix.cs]: Nit: String => string

File: src/CoreFx.Private.TestUtilities/src/System/PlatformApis.Unix.cs [hunk 3]
```diff
@@ -236,6 +237,13 @@ private static DistroInfo NormalizeDistroInfo(DistroInfo distroInfo)
...
+            {
+                distroInfo.VersionId = (new Version(Int32.MaxValue, Int32.MaxValue)).ToString();
```
Comment (stephentoub) [inline src/CoreFx.Private.TestUtilities/src/System/PlatformApis.Unix.cs]: Nit: the parens around `new Version(...)` aren't necessary

File: src/CoreFx.Private.TestUtilities/src/System/PlatformApis.Unix.cs (hunk 1 above)
Comment (danmosemsft) [reply src/CoreFx.Private.TestUtilities/src/System/PlatformApis.Unix.cs]: As noted in the linked issue, the man pages mention this file may be used as a fallback in the absence of etc/os-release. I do not think we need to do anything unless this emerges as a problem.

**Comment by ViktorHofer** [comment]
This change is problematic as PlatformApis.Unix.cs is a copy from https://github.com/dotnet/core-setup/blob/master/src/managed/Microsoft.DotNet.PlatformAbstractions/Native/PlatformApis.cs and I'm going to remove the manual copy as we don't need it anymore. Do you think we should make the same change there? If not then I will just make sure that the callee gets this change in CoreFx.Private.TestUtilities.

**Comment by bartonjs** [comment]
@ViktorHofer The reason for this change is our version detection was throwing on Debian 10.  If core-setup isn't throwing on Debian 10 then we're good to unify; if they are they presumably need some sort of fix :smile:. (Maybe it'll have a version number when it releases (current schedule prediction: next Friday))

**Comment by ViktorHofer** [comment]
Ok thanks for letting me know. I transported the fix over here https://github.com/dotnet/corefx/pull/36207/commits/457c81c299a0bad384cc91e11616eac71a1f875a which moves the detection from PlatformApis to PlatformDetection.Unix.cs.

**Pull request merged**